### PR TITLE
Enable configuring minHeight for LinearProgressIndicator and update default to match spec

### LINKED ISSUE #55481: Linear Progress Indicator does not match the material spec
The [material spec](https://material.io/components/progress-indicators#specs) calls for a height of 4dp but it is currently set to 6dp (with no way to change it).

### PR DESCRIPTION
**Pull request by rami-a**
## Description

The linear progress indicator's `minHeight` is hardcoded to 6 and unable to modified. This change enables the modification of the `minHeight` as well as sets the default to 4 to match the [material spec](https://material.io/components/progress-indicators#specs)

## Related Issues

Closes #55481

## Tests

I added the following tests:

* A test to check that customizing the `minHeight` works.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by JoseAlba — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 1]
```diff
@@ -236,6 +238,7 @@ class LinearProgressIndicator extends ProgressIndicator {
     double value,
     Color backgroundColor,
     Animation<Color> valueColor,
+    this.minHeight = 4.0,
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/progress_indicator.dart:241]: assert(minHeight == null || minHeight > 0) ?

Comment (clocksmith) [reply packages/flutter/lib/src/material/progress_indicator.dart:241]: (if its null, it can default to 4.0 when using it)

**Review by clocksmith — APPROVED** [review]

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**